File: Data Structures & Algorithms/palindromic-substrings/submission-1.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int n = s.size();
        vector<vector<int>> isPalin(n+1,vector<int> (n+1,0));
        int ans = 0;
        for(int i = 0;i<n;i++){
            isPalin[i][i] = 1;
            ans++;
        }
        for(int i = n-1;i>=0;i--){
            for(int j = i+1;j<n;j++){
                if(s[i] == s[j]){
                    if(j-i+1 <= 2){
                        isPalin[i][j] = 1;
                        ans++;
                    }
                    else{
                        if(isPalin[i+1][j-1] == 1){
                            ans++;
                            isPalin[i][j] = 1;
                        }
                    }
                }
            }
        }
      
        return ans;
    }
};

File: Data Structures & Algorithms/reverse-integer/submission-1.java
class Solution {
    public int reverse(int x) {
        boolean neg = false;
        if(x < 0) neg = true;

        long temp = Math.abs(x);
        System.out.println(temp);
        System.out.println(Integer.MAX_VALUE);
        long ans = 0;
        long t = 10;
        while(temp > 0){
            long dig = temp%t;
            temp /= t;
            ans = ((ans*10)+dig);
        }
        if(neg){
            ans *= -1;
        }
        System.out.println(ans);
        if(ans > (long)Integer.MAX_VALUE){
            return 0;
        }
        if(ans < Integer.MIN_VALUE){
            return 0;
        }
        return (int)ans;
    }
}
